### ISSUE TITLE
Device with same Device Profile data on same single dashboard

### ISSUE THREAD

**Issue by debacs83** [thingsboard/thingsboard#7842]
**Component**

 * Dashboard 
 * Widget

**Description**
I have a number of devices with same device type (Device Profile). I have made a list of devices with their attributes (latest telemetry values) in a table. 

I know how to do by clicking any particular row of the table to move to a new dashboard and add widgets to show the device values. 
In my previous build, the device ID (device name) and device profiles are different. But now I have more than 10 devices with same device profile. So, instead of opening a new Dashboard for each row (each devices), I want to open a single Dashboard, add a set of widgets and by clicking rows of the table it will redirect to the same Dashboard with same set of widgets, BUT the values on the widget should show the respective Devices' data. In IoT there might be 100s of devices of same type, so naturally we should not open 100s of new dashboards. 

Is it possible to do it in Community Edition? If so, could anyone please tell how to do it?

Best regards,
Debasish

**Environment**

 * OS:  Linux 20.04
 * ThingsBoard: Community Edition 3.4.1 
 * Browser: Google chrome, Edge, Firefox

**Comment by devaskim** [comment]
Try to use [dashboard filters](https://thingsboard.io/docs/user-guide/advanced-filters/). But to get this working you need to:
1) add server attribute to each device with its profile id (use rule chains to do it automatically, for example, on device creation)
2) once user click on a row custom action triggers to get profile id of clicked device and update server attribute of logged-in user, for example, called `DeviceProfileFilter`
3) create dashboard filter that will compare attribute from step 1) with dynamically updating user's attribute from step 2)

**Comment by debacs83** [comment]
I will try tomorrow and if get stuck will post here.

**Comment by debacs83** [comment]
It seems working. But without using Filtering. Not sure if it is the proper way. Will explain later what I did.

**Comment by debacs83** [comment]
What I did:

1. Set all the Devices' Device Profile as 'wt-920' (device model num). 
2. Set Alias name as 'air-tester' for all the devices. 
   Alias name: air-tester
   Filter type: Device type
   Device type: wt-920

3. Also set aliases for each individual devices, with respective Alias names, such as:
    Alias name: select_AT780, select_AT790, select_AT791 ....etc.
    Filter type: Entity from dashboard state 
    State entity parameter name:  By default
    Default state entity: Did not select any 

I have created (1) a Table and in the same page, (2) a map (open street map). 
Table consists of every device with their corresponding telemetry values. 
On the map the device positions are set up by 'server attributes' with fixed lat-lon values. 

In the main Dashboard where the table and map is there, in the Datasource, Type: Entity -> Entity alias as 'air-tester' (device profile name) and selected 'keys' from the telemetry values (dev_id, rssi, gw_lat, gw_lon etc.) 

Clicking either the table_row, or the pin in the map for the devices, it will move to another new dashboard, where it shows the selected device's information in a table and the corresponding Timeseries Line Chart below (two widgets in one page). 
Here, while selecting Datasource, Type: Entity, Entity alias: select_AT790 or select_AT791, or select_791  etc., irrespective of the alias of the devices, it is actually going to the correct pages for the devices. 

So I actually created ONE dashboard for all the devices (as I wanted) and even if I select alias of other device name, it is showing the correct device name in the table and the corresponding timeseries values on the Line Chart. 

Seems it is working as I wished. But feeling uncomfortable as it is working without knowing how. Attaching some screen-shots. You can see that even if I select 'any' of the devices' alias name name, the page is correctly showing the page I have selected in the main page. 

Sorry, it might be confusing even with the screen-shots. So if any other information should I need to provide, please tell me.

[TB_Dashboard_5thJan23.zip](https://github.com/thingsboard/thingsboard/files/10350097/TB_Dashboard_5thJan23.zip)

**Comment by pratikvaishya1512** [comment]
Hello, 

Can multiple Devices has same Device profile name?

For ex- Device 1, Device 2, Device 3 has Device profile name Test Profile.

If yes, the rule chain for that will also be one or different?


The reason for this query is because we are facing issue in Triggerd and Cleared alerts for multiple devices with same device profile name.


Any help please?

**Comment by ronald-cr** [comment]
Yes, you can have one device profile for multiple devices. The benefit is that you can have one rule chain for all devices (without copying rule chain logic).
If you need multiple rule chains for each device than you will need a device profile for each device, I guess.

**Comment by Ultrazombie** [comment]
Closed due to inactivity
